### ISSUE TITLE
GeneActivity() with merged object does not preserve gene order

### ISSUE THREAD

**Issue by mass-a** [timoast/signac#291]
Hello,
thank you for the wonderful package.

I have run into a possible bug when running the function `GeneActivity` over a merged object. I am running Signac version 1.0.9003.

I followed your [Merging objects vignette](https://satijalab.org/signac/articles/merging.html) (using only the *pbmc500* and *pbmc1k* objects for simplicity) all the way through calculating combined peaks, fragment objects and peak counts. Then I generated the two Seurat objects with the Chromatin assays and merged the two objects:
```
pbmc500_assay <- CreateChromatinAssay(pbmc500.counts, fragments = frags.500)
pbmc500 <- CreateSeuratObject(pbmc500_assay, assay = "ATAC")
pbmc1k_assay <- CreateChromatinAssay(pbmc1k.counts, fragments = frags.1k)
pbmc1k <- CreateSeuratObject(pbmc1k_assay, assay = "ATAC")

pbmc500$dataset <- 'pbmc500'
pbmc1k$dataset <- 'pbmc1k'
combined <- merge(pbmc500, pbmc1k, add.cell.ids = c("500", "1k"))
```

Then I add annotations to the combined as well as the individual ATAC objects:
```
annotations <- GetGRangesFromEnsDb(ensdb = EnsDb.Hsapiens.v75)
seqlevelsStyle(annotations) <- 'UCSC'
genome(annotations) <- "hg19"

Annotation(pbmc500) <- annotations
Annotation(pbmc1k) <- annotations
Annotation(combined) <- annotations
```

I proceed to calculating the gene activity for the merged object, and for reference also for the individual (pre-merging) objects:
```
gene.activities.500 <- GeneActivity(pbmc500, assay = "ATAC")
gene.activities.1k <- GeneActivity(pbmc1k, assay = "ATAC")
gene.activities.combined <- GeneActivity(combined, assay = "ATAC")
```

Now to the problem. You can see that the genes are not in the same order in the two pre-merging objects. For example:
``` 
rownames(gene.activities.500)[37]
```
[1] "WWC3"
```
rownames(gene.activities.1k)[37]
```
[1] "GPR143"
```
rownames(gene.activities.combined)[37]
```
[1] "WWC3"

The inferred gene expression counts for the first object correspond to those of the merged object:
```
sum(gene.activities.500["WWC3",])
```
[1] 822
```
sum(gene.activities.combined["WWC3",combined$dataset=="pbmc500"])
```
[1] 822

However, for the second object the counts do not correspond between the pre-merging and merged object:
```
sum(gene.activities.1k["WWC3",])
```
[1] 1554
```
sum(gene.activities.combined["WWC3",combined$dataset=="pbmc1k"])
```
[1] 99

Instead, it seems to have adopted the counts of the gene corresponding to the same line 37 for the other object ("GPR143" instead of "WWC3" in this example):
```
rownames(gene.activities.500)[37]
rownames(gene.activities.1k)[37]
sum(gene.activities.1k["GPR143",])
```
"WWC3"
"GPR143"
[1] 99

In summary, I believe that `GeneActivity` on a merged object combines the inferred expression count matrices for the objects assuming that genes are in the same order, which they are not. You may need to add a re-ordering of the gene names before binding the two count expression matrices, to preserve the correct pairing of counts. 

Do you agree?

-massimo

**Comment by timoast** [comment]
Thanks for the detailed bug report, I've just pushed a fix for this to the develop branch
